code/modules/chemistry/Chemistry-Dispenser.dm

```diff
@@ -239,6 +250,9 @@ TYPEINFO(/obj/machinery/chem_dispenser)
 
 	proc/take_damage(var/damage_amount = 5)
 		src.health -= damage_amount
+		if (src.health < 100)
+			if(prob(100-src.health))
```

Comment:
This means it almost never triggers with high damage bullets, why the `prob`?

Reply:
The behavior I want is for chem dispensers to have a higher and higher chance to break over time, without immediately breaking due to e.g. a single potassium and water mix.

I added a secondary damage roll for high-damage changes, which should handle big guns like sniper rifles more effectively. Happy to change it again if you think it needs it.